frontend/src/pages/application/CalculatePrice.js

```diff
@@ -0,0 +1,28 @@
+import {getDistance} from "geolib";
+
+export default function calculateItemPrice(item, taxes, distance, zoneId) {
+
+    let taxId = taxes.find(t => t.stateDto.id == zoneId);
+    let res = item.cost * (1 + taxId.percentage / 100) + (distance * item.category.taxRate);
+    return res.toFixed(2);
+};
+
+
+export function recalculateItems(items, taxes, distance, zoneId) {
+
+    items.forEach(i => {
+        i.price = calculateItemPrice(i, taxes, distance, zoneId);
+    });
+    return items;
```

Comment:
I would use here map instead of foreach and return the result of the map function